### ISSUE TITLE
Defeating the Phantom Knights doesn't always grant the advancement

### ISSUE THREAD

**Issue by Balnazaar** [TeamTwilight/twilightforest#718]
Forge version: 
Twilight Forest version: Latest
Link to crash log: Not crashing

Steps to reproduce:
1. Attempt to progress through the mod.
2. Kill all bosses up to hydra
3. Kill Phantom knight(multiple times)
4. Go into creative and attempt to kill Ur Ghast, fail, attempt to spawn in the tears of fire, and watch as nothing happens
What I expected to happen:
An achievement allowing me to enter the tower to kill the Ur Ghast
What happened instead:
About 2 hours of me killing the Phantom knight before realizing it was bugged

**Comment by bs2609** [comment]
Please post actual version numbers, "latest" is not specific enough to be useful.

If you are using version 3.9.851, then this is just a duplicate of #707, which has been fixed.

**Comment by EverNife** [comment]
Same for me, i can confirm that.

My players can go up to Phantom knight, but even killing him they can't go to Ur Ghast

> 
> MCVersion: 1.12.2
> Twilight Forst Version: twilightforest-1.12.2-3.9.888-universal
> Forge: 1.12.2-forge1.12.2-14.23.5.2823
> Sponge : spongeforge-1.12.2-2768-7.1.6-RC3616.jar
> 


--
Edit

Tested on 3.9.888 version

**Comment by daFritzi** [comment]
I am facing the same issue.
I used an older release (the latest before 3.9.888) the first time I killed the phantom knights.
But updating to "twilightforest-1.12.2-3.9.888-universal" and killing the knights again did not solve the issue.

I was able to kill Ur Ghast (shot it with a bow from outside the towers barrier, not sure if you should be able to do this), but I cannot interact with the chest it dropped, so that doesn't help me either.

If you need any other info, like my current mod list or anything let me know.

EDIT:
Another strange thing that happened to me was that the Hydra fight was kinda glitched. The Hydra was constantly acting like it was taking damage and did not attack me. Didn't hinder my progression though as I was able to kill it and got the blood as intended.

**Comment by ndrader** [comment]
The same issue as above, progression locked on the carminite revelations. Killed 4 Phantom Knights and the achievement has not updated and blindness is still applied when approaching Ur Ghast Tower. 
```
Twilight Forest: twilightforest-1.12.2-3.9.984-universal.jar
Forge: forge-1.12.2-14.23.5.2836-universal.jar 
SpongeForge: spongeforge-1.12.2-2838-7.1.7-RC3785.jar
Part of Dungeons and Dragons and Space Shuttle Pack version 5.7b
```


![twilight](https://user-images.githubusercontent.com/24696198/64484454-f2037f00-d1d7-11e9-9181-b4b2d130c3dd.PNG)
![twilight2](https://user-images.githubusercontent.com/24696198/64484481-3db62880-d1d8-11e9-9ca2-1e9266aa20f7.PNG)

**Comment by ndrader** [comment]
To progress the player further it was found that the vanilla minecraft command can be used:
`/advancement grant Player only twilight:knight_progress`

**Comment by hegrimmtho** [comment]
Tama has already said this somewhere, but most phantom rooms are glitched and technically out of the dungeon. There are some inside the dungeon area that can be beat and give the advancement. Hope this helps!

**Comment by EverNife** [comment]
If you need a command to progress... there is clearly a problem in it.
On multiplayer servers this turns into a problem.

**Comment by Tamaized** [comment]
The structure itself is bugged, It can generate multiple boss rooms that happen to have a chance to be outside the Structure Bounding Box. This means once that specific boss room is complete the structure's conquered flag is never set to true.

The advancement requires you be in the structure right when it gets considered conquered. You do NOT need a command to solve this on your end, simply find another boss room in the structure or if you must, find a totally new stronghold and beat that boss.

The plan to fix this will be in the structure rework but that's some time away. If you want, you can make a custom advancement on your server and remove the conquered flag from the json. Client's DO NOT need this file, only the server.

File Location to create: `{world}/data/advancements/twilightforest/progress_knights.json`

**Comment by Hyxaru** [comment]
For the time being, you can grant yourself the advancement with /advancement grant playername only twilightforest:progress_knights

**Comment by Azar2477** [comment]
That's played in Twilight Forest Version 3.9.888 in Minecraft 1.12.2.

**Comment by GizmoTheMoonPig** [comment]
this will now actually work correctly, instead of checking the structure being conquered it checks if the gang of phantoms has been defeated. This won't be backported so if you still have this issue in versions earlier than 1.18 please follow the advice given here
